lib/style_transfer.ex

```diff
@@ -0,0 +1,168 @@
+  """
+  use Membrane.Filter
+
+  defguard is_pos_integer(term) when is_integer(term) and term > 0
+
+  def_input_pad :input,
+    accepted_format:
+      %Membrane.RawVideo{pixel_format: :RGB, height: height, width: width}
+      when is_pos_integer(height) and is_pos_integer(width)
+
+  def_output_pad :output,
+    accepted_format:
+      %Membrane.RawVideo{pixel_format: :RGB, height: height, width: width}
+      when is_pos_integer(height) and is_pos_integer(width)
+
+  def_options style: [
+                required?: true,
+                spec: style(),
+                description: """
+                Style used by the element.
+
+                Can be changed by sending `{:set_style, style}` notification from a parent.
+                """
+              ],
+              batch_size: [
+                require?: false,
+                default: 1,
+                spec: non_neg_integer(),
+                description: """
+                Number of video frames passed in one batch to the model.
+
+                Can be increased in order to optimize the element at the cost of increased latency.
+
+                Default to 1.
+                """
+              ]
+
+  @styles [:candy, :kaganawa, :mosaic, :mosaic_mobile, :picasso, :princess, :udnie, :vangogh]
+
+  @type style ::
+          unquote(
+            @styles
+            |> Bunch.Typespec.enum_to_alternative()
+          )
+
+  @spec styles() :: [style()]
+  def styles(), do: @styles
+
+  @impl true
+  def handle_init(_ctx, %{style: style} = opts) when style in @styles do
+    state = %{
+      current_style: style,
+      batch_size: opts.batch_size,
+      models: %{},
+      batch: []
+    }
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_setup(_ctx, state) do
+    models =
+      @styles
+      |> Map.new(fn style ->
+        model = get_model_path(style) |> Ortex.load()
+        {style, model}
+      end)
+
+    {[], %{state | models: models}}
+  end
+
+  defp get_model_path(style) do
+    Application.get_application(__MODULE__)
+    |> :code.priv_dir()
+    |> Path.join("#{style}.onnx")
+  end
+
+  @impl true
+  def handle_parent_notification({:set_style, style}, _ctx, state) when style in @styles do
+    {[], %{state | current_style: style}}
+  end
+
+  @impl true
+  def handle_buffer(:input, buffer, ctx, state) do
+    state = Map.update!(state, :batch, &(&1 ++ [buffer]))
+
+    if length(state.batch) >= state.batch_size do
+      flush_batch(ctx.pads.input.stream_format, state)
+    else
+      {[], state}
+    end
+  end
+
+  defp flush_batch(format, state) do
+    {batch, state} = Map.get_and_update!(state, :batch, &{&1, []})
+
+    input_tensor =
+      batch
+      |> Enum.map(&preprocess(&1.payload, format))
+      |> Nx.stack()
+      |> Nx.reshape({length(batch), 3, format.height, format.width})
+
+    out_tensor = predict(input_tensor, state)
+
+    out_buffers =
+      out_tensor
+      |> Nx.backend_transfer(EXLA.Backend)
```

Comment:
I am not sure if forcing the backend is a good approach (what if somebody has some problems with EXLA setup)?

Reply:
In this specific place `backend_transfer` is necessary, otherwise next `Nx` function would raise with exception saying that "`Ortex.Backend` does not support operation `xyz`"